Sources/MapboxMaps/Gestures/GestureHandlers/PinchGestureHandler.swift

```diff
@@ -119,18 +115,34 @@ internal final class PinchGestureHandler: GestureHandler, PinchGestureHandlerPro
                     zoom: initialZoom + zoomIncrement,
                     bearing: initialRotateEnabled ? (initialBearing + rotationInDegrees) : nil))
         case .ended, .cancelled:
-            initialPinchMidpoint = nil
+            previousPinchMidpoint = nil
             initialPinchAngle = nil
-            initialCenter = nil
             initialZoom = nil
             initialBearing = nil
             initialRotateEnabled = nil
-            delegate?.gestureEnded(for: .pinch, willAnimate: false)
+            stopDraggingIfNeeded()
+            if gestureBegan {
+                delegate?.gestureEnded(for: .pinch, willAnimate: false)
+            }
+            gestureBegan = false
         default:
             break
         }
     }
 
+    private func startDragging(with point: CGPoint) {
+        precondition(!isDragging)
```

Comment:
Do we have a policy for `assert` vs `precondition`? Would it make sense to use `assert` instead?

Reply:
I'm not aware of one. I'll follow up with you directly.